### PR TITLE
feat(ios): Wave 6 edge cases + battle test

### PR DESCRIPTION
**Pull request by seantokuzo**
## Summary

Wave 6 of the sprite-agent wiring phase — edge cases + battle test on iOS. Final wave.

- **S5 Desk overflow** — when all 8 desks are occupied, new agent sprites claim a position from a 21-slot grid on the Command Bridge floor (below the lower desk row). Tappable, `/btw`-able, and properly released on despawn.
- **S4 Disconnect / reconnect** — 1-second debounced gray-out + pulsing "~" indicator on every non-idle sprite when the WebSocket drops or enters `.reconnecting`. On reconnect: clears gray, flushes queued `/btw` messages, and re-fires `sprite.state.request` so the ViewModel reconciles any subagents that spawned/despawned during the drop. The inspector swaps the `/btw` input for a "Disconnected from relay" info panel while offline.
- **S6 Fast-complete animation** — `spawnTimestamps` + `minDisplayDurationSeconds` (1.5s) guarantee spawn + walk-to-desk + work animations run to completion before the celebration/leaving chain starts. Prevents the "flash-appear / flash-gone" UX for sub-second subagents. Reused by `handleAgentComplete`, `handleAgentDismissed`, and `handleSpriteUnlink`.
- **Scenario 9 race** — `SpriteInspectorView` now reads `currentAgent` live from the ViewModel so mode (linked/dog/idle) and send payload always reflect state at send time. Race-win: idle-at-tap, linked-at-send → routed via `sprite.message` path. Race-lose: linked-at-tap, idle-at-send → brief "Agent completed" toast, draft cleared, idle inspector replaces messaging UI.
- **Verification pass** — scenarios #2 (draft discard), #3 (pending blocks), #5 (offline queue), #7 (idle human no input), #8 (empty disabled), #10 (long message expand/collapse), M1 (single-pending gate), M2 (cross-session banner), M3 (bubble priority), S1/S2 (despawn animations) — all confirmed wired.

## Test plan

- [ ] Spawn ≥9 concurrent subagents → 8 sit at desks, rest land in overflow grid without overlap
- [ ] Tap an overflow sprite → inspector opens, /btw works as expected
- [ ] Kill the relay mid-session → sprites gray out after ~1s, "~" indicator pulses
- [ ] Quickly stop/start the relay (<1s outage) → no gray flash
- [ ] Relay reconnects with new state → gray clears, completed subagents despawn, new ones show up
- [ ] Spawn a subagent that completes in <1s → sprite shows spawn + work + celebration + poof for at least 1.5s
- [ ] Tap an idle human → no input (info panel), tap a dog → canned response, tap a linked agent → /btw
- [ ] Send /btw while offline → queued, flushes on reconnect
- [ ] Open inspector on a linked agent, wait for it to complete → "Agent completed" toast, draft cleared
- [ ] With /btw modal pending, tap same sprite again → input read-only until Cool Beans

## Notes / deviations

- S3 dismiss-from-inspector (kill subagent from relay) was not wired — would require a new relay wire type and the task explicitly forbids touching `relay/` or introducing new wire types. Skipped by design.
- Relay-side Wave 6 work (reconnect fan-out, S4 disconnected-client queue behavior) is covered by the parallel `sprite-wiring/wave6-relay` PR — no cross-cutting coordination needed from this PR.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: ios/MajorTom/Features/Office/ViewModels/OfficeViewModel.swift [hunk 1]
```diff
@@ -351,6 +398,39 @@ final class OfficeViewModel {
         return deskIndex
     }
 
+    /// Claim the first free overflow slot for an agent.
+    /// Returns nil if every overflow point is already occupied (shouldn't
+    /// happen in practice — overflow pool is 18, way beyond reasonable
```
[the remark this answers is not in the document]
Comment (seantokuzo) [reply ios/MajorTom/Features/Office/ViewModels/OfficeViewModel.swift]: Fixed in 5d8e40e — updated the comment to read "overflow pool is 21 (7×3), way beyond reasonable concurrent-subagent count".

File: ios/MajorTom/Features/Office/Views/OfficeView.swift [hunk 2]
```diff
@@ -568,6 +578,58 @@ struct OfficeView: View {
...
+
+        case .connected:
+            disconnectDebounceTask?.cancel()
+            disconnectDebounceTask = nil
+            viewModel.clearDisconnectedState()
+            if let sid = viewModel.sessionId {
+                relay?.flushQueuedSpriteMessages(for: sid)
+                relay?.requestSpriteState(for: sid)
+            }
+
```
[the remark this answers is not in the document]
Comment (seantokuzo) [reply ios/MajorTom/Features/Office/Views/OfficeView.swift:616]: Fixed in 5d8e40e — moved the debounce into OfficeViewModel (owns disconnectDebounceTask + beginDisconnectDebounce/resolveReconnect methods). The pending Task now survives view appear/disappear cycles, and OfficeView.handleConnectionStateChange just delegates. Views still observe disconnectedSpriteIds via @Observable so the diff-render still works.

File: ios/MajorTom/Features/Office/Views/SpriteInspectorView.swift [hunk 3]
```diff
@@ -535,7 +631,20 @@ struct SpriteInspectorView: View {
...
+    /// Surface a short "agent completed" banner, clear the draft, and auto-hide
+    /// after 2.5s. Triggered when the inspector's linked sprite despawned mid-type.
+    private func showAgentCompletedToast() {
+        draft = ""
+        HapticService.impact(.soft)
+        agentCompletedToast = "Agent completed — /btw isn't available anymore."
+        Task { @MainActor in
+            try? await Task.sleep(for: .seconds(2.5))
+            agentCompletedToast = nil
+        }
```
[the remark this answers is not in the document]
Comment (seantokuzo) [reply ios/MajorTom/Features/Office/Views/SpriteInspectorView.swift:658]: Fixed in 5d8e40e — stored the hide Task in @State agentCompletedToastHideTask. Each call to showAgentCompletedToast() cancels the previous one before scheduling a new 2.5s hide, with a Task.isCancelled guard before clearing the state. Same pattern as the 5s tool-bubble hold fix from PR #146.

File: ios/MajorTom/Features/Office/Scenes/OfficeScene.swift [hunk 4]
```diff
@@ -1820,6 +1820,41 @@ final class OfficeScene: SKScene {
...
+    func hideDisconnectedState(on agentId: String) {
+        agentSprites[agentId]?.hideDisconnectedState()
+    }
+
+    /// True if the named sprite is currently in the disconnected visual state.
+    /// Used by OfficeView to gate inspector messaging.
+    func isSpriteDisconnected(_ agentId: String) -> Bool {
+        agentSprites[agentId]?.isDisconnected ?? false
+    }
+
```
[the remark this answers is not in the document]
Comment (seantokuzo) [reply ios/MajorTom/Features/Office/Scenes/OfficeScene.swift]: Fixed in 5d8e40e — removed isSpriteDisconnected(_:) from OfficeScene. OfficeView already queries viewModel.disconnectedSpriteIds directly (via SpriteInspectorView.isDisconnected at line 301), so the scene-side helper was dead code.

File: ios/MajorTom/Features/Office/Models/OfficeLayout.swift [hunk 5]
```diff
@@ -92,6 +92,32 @@ struct OfficeLayout {
...
+    /// Free span below desks: y=1990–2200 (~210pt tall). Furniture (captain's
+    /// chair, tactical display, status screens) all sit at the TOP of the
+    /// module, so the lower band is clear.
+    ///
+    /// Grid: 6 columns × 3 rows = 18 slots. Sprite width is ~32pt, so 80pt
+    /// column spacing + 70pt row spacing keeps overflow sprites from overlapping.
+    static let overflowPositions: [CGPoint] = {
+        var points: [CGPoint] = []
+        let xs: [CGFloat] = [80, 160, 240, 320, 400, 480, 560]
+        let ys: [CGFloat] = [2030, 2100, 2170]
```
[the remark this answers is not in the document]
Comment (seantokuzo) [reply ios/MajorTom/Features/Office/Models/OfficeLayout.swift]: Fixed in 5d8e40e — updated the comment to read "Grid: 7 columns × 3 rows = 21 slots" to match the actual xs/ys arrays (7 x-values × 3 y-values).

File: ios/MajorTom/Features/Office/ViewModels/OfficeViewModel.swift [hunk 6]
```diff
@@ -93,6 +93,41 @@ final class OfficeViewModel {
...
+
+    /// Debounce window before a disconnect paints sprites gray. Short enough
+    /// to surface long drops, long enough to swallow WebSocket blips.
+    private static let disconnectDebounceSeconds: TimeInterval = 1
+
+    // MARK: - Overflow Placement (Wave 6 — S5)
+
+    /// Overflow positions currently claimed by agent sprites, keyed by agent id.
+    /// When an agent despawns, its position is released for the next spawn.
+    var claimedOverflowPositions: [String: CGPoint] = [:]
```
[the remark this answers is not in the document]
Comment (seantokuzo) [reply ios/MajorTom/Features/Office/ViewModels/OfficeViewModel.swift]: Fixed in 78ee10b — `claimedOverflowPositions` is now `private` (dropped `private(set)` since it is never read outside `OfficeViewModel`).

File: ios/MajorTom/Features/Office/Sprites/AgentSprite.swift [hunk 7]
```diff
@@ -912,6 +912,69 @@ final class AgentSprite: SKSpriteNode {
         progressLabel = nil
     }
 
+    // MARK: - Disconnected State (Wave 6 — S4)
+
+    /// Whether the sprite is currently displayed in the disconnected visual
+    /// treatment. Read by OfficeView / inspector to swap messaging paths for
+    /// a "relay offline" message.
```
[the remark this answers is not in the document]
Comment (seantokuzo) [reply ios/MajorTom/Features/Office/Sprites/AgentSprite.swift]: Fixed in 78ee10b — rewrote the `isDisconnected` doc to reflect its actual role: an internal idempotency guard for `showDisconnectedState()` / `hideDisconnectedState()`. Also called out explicitly that `OfficeView` and `SpriteInspectorView` consume `viewModel.disconnectedSpriteIds` instead, so nobody reads this flag externally.

File: ios/MajorTom/Features/Office/Models/OfficeLayout.swift [hunk 8]
```diff
@@ -92,6 +92,32 @@ struct OfficeLayout {
...
+    /// instead of sitting at a desk.
+    ///
+    /// Command Bridge bounds: x=0–600, y=1980–2620. Desks at y=2250 and y=2450.
+    /// Free span below desks: y=1990–2200 (~210pt tall). Furniture (captain's
+    /// chair, tactical display, status screens) all sit at the TOP of the
+    /// module, so the lower band is clear.
+    ///
+    /// Grid: 7 columns × 3 rows = 21 slots. Sprite width is ~32pt, so 80pt
+    /// column spacing + 70pt row spacing keeps overflow sprites from overlapping.
+    static let overflowPositions: [CGPoint] = {
```
[the remark this answers is not in the document]
Comment (seantokuzo) [reply ios/MajorTom/Features/Office/Models/OfficeLayout.swift:109]: Fixed in 78ee10b — PR description now reads "21-slot grid" in the S5 bullet. Code comments in `OfficeLayout.swift` ("7 columns × 3 rows = 21 slots") and `OfficeViewModel.swift` ("overflow pool is 21 (7×3)") were already updated in round 1 (5d8e40e) and remain consistent.
